util/pingpong.c

```diff
@@ -0,0 +1,2111 @@
+out:
+	fi_shutdown(ct->ep, 0);
+	return ret;
+}
+
+char *ep_name(int ep_type) {
+	char *en;
+	switch(ep_type) {
+		case FI_EP_DGRAM:	en = "dgram"; break;
+		case FI_EP_RDM:		en = "rdm"; break;
+		case FI_EP_MSG:		en = "msg"; break;
+		default:		en = "none(error)"; break;
+	}
+	return en;
+}
+
+void ft_banner_info(struct ct_pingpong *ct)
+{
+	FT_DEBUG("Running pingpong test with the %s endpoint trough a %s provider ...\n", ep_name(ct->fi->ep_attr->type), ct->fi->fabric_attr->prov_name);
+	FT_DEBUG(" * Fabric Attributes:\n");
+	FT_DEBUG("  - %-20s : %s\n", "name", ct->fi->fabric_attr->name);
+	FT_DEBUG("  - %-20s : %s\n", "prov_name", ct->fi->fabric_attr->prov_name);
+	FT_DEBUG("  - %-20s : %"PRIu32"\n", "prov_version", ct->fi->fabric_attr->prov_version);
+	FT_DEBUG(" * Domain Attributes:\n");
+	FT_DEBUG("  - %-20s : %s\n", "name", ct->fi->domain_attr->name);
+	FT_DEBUG("  - %-20s : %zu\n", "cq_cnt", ct->fi->domain_attr->cq_cnt);
+	FT_DEBUG("  - %-20s : %zu\n", "cq_data_size", ct->fi->domain_attr->cq_data_size);
+	FT_DEBUG("  - %-20s : %zu\n", "ep_cnt", ct->fi->domain_attr->ep_cnt);
+	FT_DEBUG(" * Endpoint Attributes:\n");
+	FT_DEBUG("  - %-20s : %s\n", "type", ep_name(ct->fi->ep_attr->type));
+	FT_DEBUG("  - %-20s : %"PRIu32"\n", "protocol", ct->fi->ep_attr->protocol);
+	FT_DEBUG("  - %-20s : %"PRIu32"\n", "protocol_version", ct->fi->ep_attr->protocol_version);
+	FT_DEBUG("  - %-20s : %zu\n", "max_msg_size", ct->fi->ep_attr->max_msg_size);
+	FT_DEBUG("  - %-20s : %zu\n", "max_order_raw_size", ct->fi->ep_attr->max_order_raw_size);
+}
+
+void ft_init_ct_pingpong(struct ct_pingpong *ct)
+{
+	ct->fi_pep = NULL;
+	ct->fi = NULL;
+	ct->hints = NULL;
+	ct->fabric = NULL;
+	ct->domain = NULL;
+	ct->pep = NULL;
+	ct->ep = NULL;
+	ct->txcq = NULL;
+	ct->rxcq = NULL;
+	ct->mr = NULL;
+	ct->av = NULL;
+	ct->eq = NULL;
+
+	//no_mr;
+	ct->ctx_arr = NULL;
+	ct->remote_cq_data = 0;
+
+	ct->tx_seq = 0;
+	ct->rx_seq = 0;
+       	ct->tx_cq_cntr = 0;
+	ct->rx_cq_cntr = 0;
+
+	//ct->ft_skip_mr = 0;
+	ct->ft_parent_proc = 0;
+	ct->ft_child_pid = 0;
+
+	ct->remote_fi_addr = FI_ADDR_UNSPEC;
+	ct->buf = NULL;
+	ct->tx_buf = NULL;
+	ct->rx_buf = NULL;
+
+	ct->buf_size = 0;
+       	ct->tx_size = 0;
+	ct->rx_size = 0;
+	ct->rx_fd = -1;
+	ct->tx_fd = -1;
+
+	strncpy(ct->test_name, "custom", 50);
+	ct->timeout = -1;
+	//start, end;
+
+	ct->av_attr = (struct fi_av_attr) {
+		.type = FI_AV_MAP,
+		.count = 1
+	};
+	ct->eq_attr = (struct fi_eq_attr) {
+		.wait_obj = FI_WAIT_UNSPEC
+	};
+	ct->cq_attr = (struct fi_cq_attr) {
+		.wait_obj = FI_WAIT_NONE
+	};
+
+	ct->data_default_port = 9228;
+	ct->ctrl_port = 47592;
+
+	memset(ct->ctrl_buf, '\0', sizeof(ct->ctrl_buf));
+}
+
+int main(int argc, char **argv)
+{
+	int ret, op;
+
+	ret = EXIT_SUCCESS;
+
+	struct ct_pingpong ct;
+
+	ft_init_ct_pingpong(&ct);
+	ct.opts = INIT_OPTS;
+
+	ct.hints = fi_allocinfo();
+	if (!ct.hints)
+		return EXIT_FAILURE;
+
+	while ((op = getopt(argc, argv, "h:d" CS_OPTS INFO_OPTS BENCHMARK_OPTS)) !=
```

Comment:
The `h` option does not accept a parameter. This should be `"hd"`.

It currently results in: 

```
./fi_pingpong: option requires an argument -- 'h'
```